cli/azd/cmd/infra_create.go

```diff
@@ -272,25 +276,78 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 	return nil
 }
 
-func reportDeploymentStatusInteractive(ctx context.Context, azCli tools.AzCli, env environment.Environment, showProgress func(string)) {
+func reportDeploymentStatusInteractive(ctx context.Context, azCli tools.AzCli, env environment.Environment, spinnerUpdater *spin.SpinnerUpdater, loggedResourcesPtr *map[string]bool) {
 	resourceManager := infra.NewAzureResourceManager(azCli)
 
-	operations, err := resourceManager.GetDeploymentResourceOperations(ctx, env.GetSubscriptionId(), env.GetEnvName())
+	resOps, err := resourceManager.GetDeploymentResourceOperations(ctx, env.GetSubscriptionId(), env.GetEnvName())
 	if err != nil {
 		// Status display is best-effort activity.
 		return
 	}
 
+	operations := *resOps
 	succeededCount := 0
+	newlyDeployedResources := []azureutil.DeployedAzureResource{}
+	loggedResources := *loggedResourcesPtr
 
-	for _, resourceOperation := range *operations {
+	for _, resourceOperation := range operations {
 		if resourceOperation.Properties.ProvisioningState == "Succeeded" {
 			succeededCount++
+
+			if !loggedResources[resourceOperation.Properties.TargetResource.Id] {
+				newlyDeployedResources = append(newlyDeployedResources, azureutil.DeployedAzureResource{
+					Id:                resourceOperation.Properties.TargetResource.Id,
+					ResourceName:      resourceOperation.Properties.TargetResource.ResourceName,
+					ResourceType:      resourceOperation.Properties.TargetResource.ResourceType,
+					DeployedTimestamp: resourceOperation.Properties.Timestamp,
+				})
+			}
+		}
+	}
+
+	sort.Sort(azureutil.DeployedAzureResourceByTimestamp(newlyDeployedResources))
+
+	for i := range newlyDeployedResources {
```

Comment:
Consider moving all of these loops and process on top of the operations into `ResourceManager` to it will be easier to refactor this with the upcoming provider support.

Reply:
I'll take a look at your draft PR to see what changes we should make.

Reply:
Refactored out the display logic into its own class. It should be a lot more reusable now. The display logic mostly keeps track of succeeded resources, and also does formatting logic.

The one additional change I think would simplify it (based on the direction of your PR) is to have the display accept the deployment operations directly `ReportProgress(operations []tools.AzCliResourceOperation)`. I think this would line up well if we were to use channels to send the operations as a progress report back to `infra_create.go`. Let me know if you want me to make this change.